httpd/http_server.c

```diff
@@ -0,0 +1,333 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "daemon/common.h"
+#include "http_server.h"
+#include "h2o.h"
+
+#include "h2o_utils.h"
+
+static h2o_globalconf_t config;
+static h2o_context_t ctx;
+static h2o_accept_ctx_t accept_ctx;
+
+#define CONTENT_JSON_UTF8 H2O_STRLIT("application/json; charset=utf-8")
+#define CONTENT_TEXT_UTF8 H2O_STRLIT("text/plain; charset=utf-8")
+#define NBUF_INITIAL_SIZE_RESP (4096)
+#define API_V1_PREFIX "/api/v1/"
+#define HOST_SELECT_PREFIX "/host/"
+
+#define HTTPD_CONFIG_SECTION "httpd"
+#define HTTPD_ENABLED_DEFAULT false
+
+static void on_accept(uv_stream_t *listener, int status)
+{
+    uv_tcp_t *conn;
+    h2o_socket_t *sock;
+
+    if (status != 0)
+        return;
+
+    conn = h2o_mem_alloc(sizeof(*conn));
+    uv_tcp_init(listener->loop, conn);
+
+    if (uv_accept(listener, (uv_stream_t *)conn) != 0) {
+        uv_close((uv_handle_t *)conn, (uv_close_cb)free);
+        return;
+    }
+
+    sock = h2o_uv_socket_create((uv_stream_t *)conn, (uv_close_cb)free);
+    h2o_accept(&accept_ctx, sock);
+}
+
+static int create_listener(const char *ip, int port)
+{
+    static uv_tcp_t listener;
+    struct sockaddr_in addr;
+    int r;
+
+    uv_tcp_init(ctx.loop, &listener);
+    uv_ip4_addr(ip, port, &addr);
```

Comment:
What about `IPV6` addresses? Should we check IP version and also call `uv_ip6_addr` when it is necessary?
I am studying the code from top to bottle, if I find answer in next lines I will remove this comment.

Reply:
Yes, we should handle IPV6 addresses here.

Reply:
@underhood I did not find any place addressing IPV6, this was the motive I did not remove the initial comment.
@Ferroin thank you to share your opinion.

Reply:
i added ip6 decode if ip4 decode fails